Dangerfile

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+def release_branch?
+  danger.github.branch_for_base.start_with?('release/') || danger.github.branch_for_base.start_with?('hotfix/')
+end
+
+def main_branch?
+  danger.github.branch_for_base == 'trunk'
+end
+
+def wip_feature?
+  has_wip_label = github.pr_labels.any? { |label| label.include?('WIP') }
+  has_wip_title = github.pr_title.include?('WIP')
+
+  has_wip_label || has_wip_title
+end
+
+return if github.pr_labels.include?('Releases')
+
+github.dismiss_out_of_range_messages
+
+manifest_pr_checker.check_gemfile_lock_updated
+
+labels_checker.check(
+  required_labels: [//],
+  required_labels_error: 'PR requires at least one label.'
+)
+
+view_changes_need_screenshots.view_changes_need_screenshots
+
+pr_size_checker.check_diff_size(
+  file_selector: ->(path) { !path.include?('/src/test') },
+  max_size: 300
+)
+
+android_unit_test_checker.check_missing_tests
+
+# skip check for draft PRs and for WIP features unless the PR is against the main branch or release branch
+milestone_checker.check_milestone_due_date(days_before_due: 4) if !github.pr_draft? && (!wip_feature? || (release_branch? || main_branch?))
```

Comment:
What do you think of using positive logic and using the same wording as the comment in the WIP... logic?

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 4) unless github.pr_draft? || (wip_feature? && !(release_branch? || main_branch?))
```

Reply:
That's a great suggestion -- I actually had a lot of trouble parsing the original rule in Peril, this improves a lot the readability, going together with the comment.